src/atlannot/evaluation.py

```diff
@@ -0,0 +1,221 @@
+        intersection over union of that given region id as values.
+    """
+    results = {}
+    if region_ids is None:
+        labels = np.unique(vol_true)
+        for label in labels:
+            results[label] = iou_score(vol_true, vol_pred, k=label)[0]
+        return results
+
+    for region_id in region_ids:
+        results[region_id] = iou_score(vol_true, vol_pred, k=region_id)[0]
+
+    return results
+
+
+def compute_region_entropy(nissl, atlas, label_value, bins=100):
+    """Compute the entropy of densities of Nissl for a given label value in the atlas.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+    label_value: int
+        Label Value to consider.
+    bins: int
+        Number of bins used to compute the histogram of the densities.
+
+    Returns
+    -------
+    region_entropy: float
+        Entropy of the densities of Nissl at a given region value.
+    """
+    prob_region, _ = np.histogram(nissl[atlas == label_value], bins=bins)
+    region_entropy = entropy(prob_region)
+    return region_entropy
+
+
+def compute_entropies(nissl, atlas):
+    """Compute entropies of Nissl densities.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+
+    Returns
+    -------
+    brain_entropy: float
+        Entropy of the densities of Nissl for the entire brain.
+
+    conditional_entropy: float
+        Conditional entropy of the densities of Nissl depending on the brain regions.
+    """
+    # 1: For the entire brain
+    prob_entire_brain, _ = np.histogram(nissl[atlas != 0], bins=100)
+    brain_entropy = entropy(prob_entire_brain)
+
+    # 2. Computation of the conditional entropy
+    n_pixels = (atlas != 0).sum()
+    label_values, count_values = np.unique(atlas, return_counts=True)
+    all_region_entropy = []
+    for label, count in zip(label_values, count_values):
+        all_region_entropy.append(compute_region_entropy(nissl, atlas, label) * count)
+
+    conditional_entropy = np.sum(all_region_entropy) / n_pixels
+    return brain_entropy, conditional_entropy
+
+
+def evaluate(
+    atlas,
+    nissl,
+    reference,
+    region_meta,
+    regions_to_evaluate=REGIONS_TO_EVALUATE,
+):
+    """Evaluate the atlas.
+
+    Parameters
+    ----------
+    atlas: np.ndarray
+        Atlas to evaluate.
+    nissl: np.ndarray
+        Corresponding Nissl Volume.
+    reference: np.ndarray
+        Reference atlas.
+    region_meta: atlannot.atlas.RegionMeta
+        Region Meta containing all the information concerning the labels.
+    regions_to_evaluate: dict[str, list[int]]
+        Regions to evaluate.
+
+    Returns
+    -------
+    results: dict[str, Any]
+        Dictionary containing the results of the evaluation.
+    """
+    results = {}
+    for name, region_ids in regions_to_evaluate.items():
```

Comment:
(optional)

If you're fancy you can refactor even more to focus on the evaluation of a single region. One could consider replacing dictionaries with a dataclass for the result:
```python
def evaluate_region(
    atlas,
    nissl,
    reference,
    region_meta,
    region_ids,
) -> RegionEvaluation:
    ...
    evaluation = RegionEvaluation(
        region_ids=region_ids,
        region_descendants=desc,
        region_levels=...,
        global_jaggedness=...,
        region_jaggedness=...,
        ...
    )
    ...
    return evaluation
```
The whole structure kind of depends on how the result will be used later on.

Reply:
I separated the evaluation of a given region from the entire evaluation. What do you think of this ? 759d00f

Reply:
Looks nice, thanks! The return type `dict[str, list[int] | dict[str, float | list[float]]]` looks a bit nasty. If it turn out the detailed value types are not so useful then one might consider just writing `dict[str, Any]`.